wg-api/spec-documents/scroll-view.md

```diff
@@ -0,0 +1,281 @@
+
+#### `scrollView.setVerticalScrollBarMode(mode)` _Experimental_
+
+* `mode` string - Can be `disabled`, `enabled-but-hidden`, `enabled`. Default is `enabled`.
+
+Controls how the vertical scroll bar appears and functions.
+
+#### `scrollView.getHorizontalScrollBarMode()` _Experimental_
+
+Returns `string` - horizontal scrollbar mode.
+
+#### `scrollView.getVerticalScrollBarMode()` _Experimental_
+
+Returns `string` - vertical scrollbar mode.
+
+#### `scrollView.setHorizontalScrollElasticity(elasticity)` _macOS_ _Experimental_
+
+* `elasticity` string - Can be `automatic`, `none`, `allowed`. Default is `automatic`.
+
+The scroll view’s horizontal scrolling elasticity mode.
+A scroll view can scroll its contents past its bounds to achieve an elastic effect. 
+When set to `automatic`, scrolling the horizontal axis beyond its document
+bounds only occurs if the document width is greater than the view width,
+or the vertical scroller is hidden and the horizontal scroller is visible.
+* `automatic` - Automatically determine whether to allow elasticity on this axis.
+* `none` - Disallow scrolling beyond document bounds on this axis.
+* `allowed` - Allow content to be scrolled past its bounds on this axis in an elastic fashion.
+
+#### `scrollView.setVerticalScrollElasticity(elasticity)` _macOS_ _Experimental_
+
+* `elasticity` string - Can be `automatic`, `none`, `allowed`. Default is `automatic`.
+
+The scroll view’s vertical scrolling elasticity mode.
+
+#### `scrollView.getHorizontalScrollElasticity()` _macOS_ _Experimental_
+
+Returns `string` - The scroll view’s horizontal scrolling elasticity mode.
+
+#### `scrollView.getVerticalScrollElasticity()` _macOS_ _Experimental_
+
+Returns `string` - The scroll view’s vertical scrolling elasticity mode.
+
+#### `view.setScrollPosition(point)` _Experimental_
+
+* `point` - The point in the `contentView` to scroll to.
+
+Scrolls the view’s closest ancestor `clipView` object so a point in the
+view lies at the origin of the clip view's bounds rectangle.
+
+#### `view.getScrollPosition()` _Experimental_
+
+Returns `Point`
+
+#### `view.getMaximumScrollPosition()` _Experimental_
+
+Returns `Point`
+
+#### Events
+
+#### Event: 'did-scroll' _Experimental_
+
+Returns:
+
+* `event` Event
+
+Emitted when the content view is being scrolled.
+
+#### Event: 'start-scroll' _Experimental_
+
+Returns:
+
+* `event` Event
+
+Emitted at the beginning of user-initiated scroll tracking.
+
+#### Event: 'end-scroll' _Experimental_
+
+Returns:
+
+* `event` Event
+
+Emitted at the end of scroll tracking.
+
+## WebBrowserView
+
+A `WebBrowserView` is the alternative for `BrowserView`. It extends `BaseView`, so it can be used as part of view's hierarchy.
+
+Process: **Main**
+
+### `new WebBrowserView([options])` _Experimental_
+
+* `options` Object (optional)
+  * `webPreferences` (optional)
+
+### Instance Properties
+
+#### `view.webContents` _Experimental_
+
+A `WebContents` object owned by this view.
+
+## Rollout Plan
```

Comment:
> Can we want to hide this behind a command line switch at first? This would make it more explicit for users that this API is subject to change.
Would you like to hide  all APIs (BaseView, ScrollView, WebBrowserView) behind a command line switch? I think that would be possible.
> • Document existing View APIs that landed from @zcbenz
> • Next-gen view API replacing legacy BrowserView 
Looks reasonable for me.
> ◦ For stability: how do we maintain backward-compatibility for the deprecated BrowserView? Can we make it a JS API layer around the new WebBrowserView?
We could do it in that way. The previously proposed WrapperBrowserView would provide somehow  backward-compatibility for the BrowserView, we could add 'browserView' to the options of WebBrowserView. We would not be able add BrowserView to the hierarchy of BaseView directly.

Reply:
Added the few points in 8418e0413a4cccc2258add477052713d647a9300